File: python/sample-authentication-code/getAccessToken.py
"""
This snippet highlights how to obtain an OAuth access token using Python.
You will provide your CLIENT_ID, CLIENT_SECRET and ACCESS_TOKEN_URL.
You can obtain these values following this guide: https://developer.payments.jpmorgan.com/quick-start
"""
import requests


def get_access_token(url, client_id, client_secret):
    response = requests.post(
        url,
        data={"grant_type": "client_credentials", "scope":"jpm:payments:sandbox"},
        auth=(client_id, client_secret),
    )
    return response.json()["access_token"]

File: go/sample-authentication-code/getAccessToken.go
package sample_authentication_code

import (
	"encoding/json"
	"fmt"
	"net/http"
	"strings"
)

func getAccessToken(url, clientID, clientSecret string) (string, error) {
	// Prepare request body
	payload := strings.NewReader("grant_type=client_credentials&scope=jpm:payments:sandbox")

	// Create HTTP client
	httpClient := &http.Client{}

	// Create HTTP request
	req, err := http.NewRequest("POST", url, payload)
	if err != nil {
		return "", err
	}
	req.SetBasicAuth(clientID, clientSecret)
	req.Header.Add("Content-Type", "application/x-www-form-urlencoded")

	// Send request
	resp, err := httpClient.Do(req)
	if err != nil {
		return "", err
	}
	defer resp.Body.Close()

	// Decode response
	var responseMap map[string]interface{}
	err = json.NewDecoder(resp.Body).Decode(&responseMap)
	if err != nil {
		return "", err
	}

	// Extract access token from response
	accessToken, ok := responseMap["access_token"].(string)
	if !ok {
		return "", fmt.Errorf("access token not found in response")
	}

	return accessToken, nil
}
